.github/workflows/agent-packaging.yml

```diff
@@ -0,0 +1,104 @@
+name: Release Agent v3
+
+on:
+  workflow_call:
+    inputs:
+      packageVersion:
+        type: string
+        default: "3.0.0"
+      packageBuildNo:
+        type: string
+        default: "1"
+      tagRelease:
+        type: boolean
+        default: false
+      releaseBranch:
+        type: string
+        required: true
+      signingKeyFile:
+        type: string
+        required: true
+      uploadUrl:
+        type: string
+        default: "https://up-ap.nginx.com"
+    secrets:
+      INDIGO_GPG_AGENT:
+
+
+env:
+  NFPM_VERSION: 'v2.35.3'
+  GOPROXY: "https://${{ secrets.ARTIFACTORY_USER }}:${{ secrets.ARTIFACTORY_TOKEN }}@azr.artifactory.f5net.com/artifactory/api/go/f5-nginx-go-dev"
+
+permissions:
+  contents: read
+  id-token: write
+
+defaults:
+  run:
+    shell: 'bash -Eeo pipefail -x {0}'
+
+jobs:
```

Comment:
For other workflows we also have a "sanity check" job that runs on public runners - and makes sure the workflow is instantiated from our orgs and through allowed events on known branches.  See e.g. https://github.com/nginx/ci-self-hosted/blob/4253880f6fc0885e60030c71c1004e4cff6d22bb/.github/workflows/nginx-buildbot.yml#L14

While it's probably something that isnt strictly necessary, it's an additional check to make sure our self-hosted runners wont run the jobs they're not supposed to run.